src/lib.rs

```diff
@@ -67,202 +77,130 @@ world.spawn((
+and register it as an input context. We would then define new actions specific to vehicle control,
+such as `Accelerate` and `Brake`, and associate them with the `InVehicle` context entity.
 
-Some bindings are very common. It would be inconvenient to bind WASD keys and analog sticks manually, like in the example above,
-every time. To solve this, we provide [presets](crate::preset) - structs that implement [`SpawnableList`] and store bindings that
-will be spawned with predefined modifiers. To spawn them, you need to to call [`SpawnRelated::spawn`] implemented for [`Bindings`]
-directly instead of the [`bindings!`] macro.
+If we wanted to add another action to the `OnFoot` context, such as `Crouch`, we would define a new action type `Crouch`
+and associate it with the `OnFoot` context for our player entity.
 
-For example, you can use [`Cardinal`] and [`Axial`] presets to simplify the example above.
+And if we wanted to add a new key binding for the `Jump` action, such as the "J" key,
+we would simply add a new binding to the existing `Jump` action on our player entity.
 
-```
-# use bevy::prelude::*;
-# use bevy_enhanced_input::prelude::*;
-# let mut world = World::new();
-world.spawn((
-    Player,
-    actions!(Player[
-        (
-            Action::<Movement>::new(),
-            DeadZone::default(),
-            SmoothNudge::default(),
-            Bindings::spawn((
-                Cardinal::wasd_keys(),
-                Axial::left_stick(),
-            )),
-        ),
-    ]),
-));
-# #[derive(Component)]
-# struct Player;
-# #[derive(InputAction)]
-# #[action_output(Vec2)]
-# struct Movement;
-```
+These patterns make it easy to manage complex input schemes in a structured but flexible way,
+and support complex scenarios like multiple players, different gameplay states, customizable controls,
+and computer-controlled entities that take the same actions as players.
 
-You can also assign custom bindings or attach additional modifiers, see the [preset] module for more details.
+## More sophisticated input patterns
 
-## Input conditions
+While we can bind actions directly to input sources like buttons, further customization or preprocessing is often needed.
+Not all inputs are "buttonlike": we may want to only trigger an action when a button is held for a certain duration,
+or when a joystick is moved beyond a certain threshold.
 
-Instead of hardcoded states like "pressed" or "released", all actions use an abstract [`ActionState`] component
-(which is a required component of [`Action<C>`]). Its meaning depends on the assigned [input conditions](crate::condition),
-which determine when the action is triggered. This allows you to define flexible behaviors, such as "hold for 1 second".
+There are two main ways to achieve this: using input conditions and input modifiers:
+- Input conditions define when an action is considered to be triggered based on the state of its bindings.
```

Comment:
We probably need to mention that without any input condition it works like `Down` without any actuation threshold.